### PR TITLE
Debug collector input processing

### PR DESCRIPTION
**Pull request by NicholasSpringer**
Processing of CLI input into a config that can immediately be used by the collector to collect data.
#124, #127

### CODE REVIEW

**Comment by NicholasSpringer** [comment]
> I don't think we want to be tracking binaries in this repo, can those two bins be generated when needed by the tests instead?

Yes, but then those tests would require linkers for specific architectures.

**Comment by jonlamb-gh** [comment]
> > I don't think we want to be tracking binaries in this repo, can those two bins be generated when needed by the tests instead?
> 
> Yes, but then those tests would require linkers for specific architectures.

We should be able to use rust-lld I would think.

**Comment by NicholasSpringer** [comment]
> We should be able to use rust-lld I would think.

Oh, ok. I'll look into that

File: ekotrace-debug-collector/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "ekotrace-debug-collector"
+version = "0.1.0"
+authors = ["Nicholas Springer <nicholas.m.springer@gmail.com>"]
```
Comment (jonlamb-gh) [inline ekotrace-debug-collector/Cargo.toml:4]: I think we've all been using our Auxon addresses in here, `nspringer@auxon.io`?

File: ekotrace-debug-collector/src/cli.rs [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+            }
+        );
+        assert_eq!(
+            config_from_options(options_from_str(
+                "ekotrace-debug-collector \
+                --session-id 0 \
+                --attach stm32 \
+                --interval 1s \
+                --output-file ./out \
+                --elf ./tests/example-be-elf \
```
Comment (NicholasSpringer) [inline ekotrace-debug-collector/src/cli.rs:483]: I switched to a runtime-compiled crate for testing the symbol parsing, but I think it makes sense for endianness detection to just include a small binary. It's just 600 bytes and basically just consists of the ELF header that is read to figure out endianness, so there's not any reason we'd want to change this.

**Comment by jonlamb-gh** [comment]
I think we can close this in favor of #142

**Pull request closed without merging**